iree_tests/benchmarks/sdxl/benchmark_sdxl_rocm.py

```diff
@@ -159,46 +163,145 @@ def job_summary_process(ret_value, output):
     return benchmark_mean_time
 
 def test_sdxl_rocm_benchmark(goldentime_rocm_e2e, goldentime_rocm_unet, 
-    goldentime_rocm_clip, goldentime_rocm_vae, gpu_number, rocm_chip):
-    # if the benchmark returns 1, benchmark failed
-    job_summary_lines = []
+    goldentime_rocm_clip, goldentime_rocm_vae, gpu_number, rocm_chip,
+    goldendispatch_rocm_unet, goldendispatch_rocm_clip, goldendispatch_rocm_vae,
+    goldensize_rocm_unet, goldensize_rocm_clip, goldensize_rocm_vae):
 
     # e2e benchmark
     ret_value, output = run_sdxl_rocm_benchmark(rocm_chip, gpu_number)
     benchmark_mean_time = job_summary_process(ret_value, output)
-    assert benchmark_mean_time <= goldentime_rocm_e2e, "SDXL e2e benchmark time should not regress"
+    check.less_equal(benchmark_mean_time, goldentime_rocm_e2e, "SDXL e2e benchmark time should not regress")
     mean_line = (f"E2E Benchmark Time: {str(benchmark_mean_time)} ms"
                  f" (golden time {goldentime_rocm_e2e} ms)")
-    job_summary_lines.append(mean_line)
+    e2e_mean_time = benchmark_mean_time
     logging.getLogger().info(mean_line)
 
     # unet benchmark
     ret_value, output = run_sdxl_unet_rocm_benchmark(gpu_number)
     benchmark_mean_time = job_summary_process(ret_value, output)
-    assert benchmark_mean_time <= goldentime_rocm_unet, "SDXL unet benchmark time should not regress"
+    check.less_equal(benchmark_mean_time, goldentime_rocm_unet, "SDXL unet benchmark time should not regress")
     mean_line = (f"Scheduled Unet Benchmark Time: {str(benchmark_mean_time)} ms"
                  f" (golden time {goldentime_rocm_unet} ms)")
-    job_summary_lines.append(mean_line)
+    unet_mean_time = benchmark_mean_time
     logging.getLogger().info(mean_line)
 
+    # unet compilation stats check
+    with open(f"{scheduled_unet_dir}/compilation_info.json", 'r') as file:
+        comp_stats = json.load(file)
+    unet_dispatch_count = int(comp_stats["stream-aggregate"]["execution"]["dispatch-count"])
+    check.less_equal(unet_dispatch_count, goldendispatch_rocm_unet, "SDXL scheduled unet dispatch count should not regress")
+    compilation_line = (f"Scheduled Unet Dispatch Count: {unet_dispatch_count}"
+                        f" (golden dispatch count {goldendispatch_rocm_unet})")
+    logging.getLogger().info(compilation_line)
+
+    module_path = f"{scheduled_unet_dir}/model_gpu_rocm_real_weights.vmfb"
+    binary_size = Path(module_path).stat().st_size
+    check.less_equal(binary_size, goldensize_rocm_unet, "SDXL scheduled unet binary size should not get bigger")
```

Comment:
Similar comment here - increasing binary size can improve performance (e.g. if including more specialized kernels, or generating more unrolled loops). At the same time, decreasing binary size can improve performance too (e.g. folding sequences of ops or reducing the number of barriers in a program)

@MaheshRavishankar should this just check the exact value? Or log the value but not check it against a reference?